pystac/extensions/sar.py

```diff
@@ -353,27 +567,45 @@ def migrate(
         if version < "0.9":
             # Some sar fields became common_metadata
             if (
-                "sar:platform" in obj["properties"]
+                PREFIX + "platform" in obj["properties"]
                 and "platform" not in obj["properties"]
             ):
-                obj["properties"]["platform"] = obj["properties"]["sar:platform"]
-                del obj["properties"]["sar:platform"]
+                obj["properties"]["platform"] = obj["properties"].pop(
+                    PREFIX + "platform"
+                )
 
             if (
-                "sar:instrument" in obj["properties"]
+                PREFIX + "instrument" in obj["properties"]
                 and "instruments" not in obj["properties"]
             ):
-                obj["properties"]["instruments"] = [obj["properties"]["sar:instrument"]]
-                del obj["properties"]["sar:instrument"]
+                obj["properties"]["instruments"] = [
+                    obj["properties"].pop(PREFIX + "instrument")
+                ]
 
             if (
-                "sar:constellation" in obj["properties"]
+                PREFIX + "constellation" in obj["properties"]
                 and "constellation" not in obj["properties"]
             ):
-                obj["properties"]["constellation"] = obj["properties"][
-                    "sar:constellation"
+                obj["properties"]["constellation"] = obj["properties"].pop(
+                    PREFIX + "constellation"
+                )
+
+            # Some SAR fields changed property names
+            if (
+                PREFIX + "type" in obj["properties"]
+                and PRODUCT_TYPE_PROP not in obj["properties"]
+            ):
+                obj["properties"][PRODUCT_TYPE_PROP] = obj["properties"].pop(
+                    PREFIX + "type"
+                )
+
+            if (
+                PREFIX + "polarization" in obj["properties"]
+                and POLARIZATIONS_PROP not in obj["properties"]
+            ):
+                obj["properties"][POLARIZATIONS_PROP] = [
+                    obj["properties"].pop(PREFIX + "polarization")
                 ]
```

Comment:
@m-mohr Can you verify that this is the right way to handle these migrations for the changes in v0.9.0?

Reply:
I'm not 100% fluent in Python. sar:constellation/instrument(s)/platform is looking good.

What is this the code for type and polarization(s) doing?

Reply:
For `sar:type` this is converting the `sar:type` property to a `sar:product_type` property. For `sar:polarization` it takes the `sar:polarization` value and puts it into a `sar:polarizations` list.

So this:

```json
{
    "sar:type": "SSC",
    "sar:polarization": "HH"
}
```

would be converted to this:

```json
{
    "sar:product_type": "SSC",
    "sar:polarizations": ["HH"]
}
```

Reply:
That looks good to me.